src/include/compat.h

```diff
@@ -195,24 +215,29 @@ struct cpu_set_t;
     pthread_set_name_np(thread, name);          \
     0; })
 #else
-  /* compiler warning free success noop */
-  #define ceph_pthread_setname(thread, name) ({ \
-    int __i = 0;                              \
-    __i; })
+  #define ceph_pthread_setname pthread_setname_noop_helper
 #endif
 
-#if defined(HAVE_PTHREAD_GETNAME_NP)
+#if defined(_WIN32) && defined(__clang__) && \
+    !defined(_LIBCPP_HAS_THREAD_API_PTHREAD)
+  #define ceph_pthread_getname pthread_getname_noop_helper
+#elif defined(HAVE_PTHREAD_GETNAME_NP)
   #define ceph_pthread_getname pthread_getname_np
 #elif defined(HAVE_PTHREAD_GET_NAME_NP)
   #define ceph_pthread_getname(thread, name, len) ({ \
     pthread_get_name_np(thread, name, len);          \
     0; })
 #else
-  /* compiler warning free success noop */
-  #define ceph_pthread_getname(thread, name, len) ({ \
-    if (name != NULL)                              \
-      *name = '\0';                                \
-    0; })
+  #define ceph_pthread_getname pthread_getname_noop_helper
+#endif
+
+#if defined(_WIN32) && defined(__clang__) && \
+    !defined(_LIBCPP_HAS_THREAD_API_PTHREAD)
+  #define ceph_pthread_kill(thread, signal) \
+    ceph_pthread_kill_unsupported(thread, signal)
```

Comment:
Rename + simplification (outstanding from the previous iteration), here and below:
```suggestion
  #define ceph_pthread_kill pthread_kill_unsupported_helper
```